apollo-router/src/configuration/experimental.rs

```diff
@@ -1,71 +1,101 @@
+            .preview
+            .iter()
+            .map(|(used_conf, discussion_link)| format!("\t- {used_conf}: {discussion_link}"))
+            .collect();
+        println!(
+            "List of all preview configurations with related GitHub discussions:\n\n{}",
+            available_confs_str.join("\n")
+        );
+    }
+
+    pub(crate) fn log_experimental_used(&self, conf: &Value) {
+        let used_experimental_conf = get_configurations(conf, "experimental");
+
         let needed_discussions: Vec<String> = used_experimental_conf
             .into_iter()
             .filter_map(|used_exp_conf| {
-                available_discussions
+                self.experimental
                     .get(&used_exp_conf)
                     .map(|discussion_link| format!("\t- {used_exp_conf}: {discussion_link}"))
             })
             .collect();
         if !needed_discussions.is_empty() {
             tracing::info!(
-                r#"You're using some "experimental" features (configuration prefixed by "experimental_" or contained within an "experimental" section), we may make breaking changes in future releases.
-To help us design the stable version we need your feedback, here is a list of links where you can give your opinion:
+                r#"You're using some "experimental" features (configuration prefixed by "experimental_" or contained within an "experimental" section). We may make breaking changes in future releases.
+To help us design the stable version we need your feedback. Here is a list of links where you can give your opinion:
 
 {}
 "#,
                 needed_discussions.join("\n")
             );
         }
     }
-}
 
-fn get_experimental_configurations(conf: &Value) -> Vec<String> {
-    let mut experimental_fields = Vec::new();
-    visit_experimental_configurations(conf, &mut experimental_fields);
+    pub(crate) fn log_preview_used(&self, conf: &Value) {
+        let used_preview_conf = get_configurations(conf, "preview");
+
+        let needed_discussions: Vec<String> = used_preview_conf
+            .into_iter()
+            .filter_map(|used_conf| {
+                self.experimental
+                    .get(&used_conf)
+                    .map(|discussion_link| format!("\t- {used_conf}: {discussion_link}"))
+            })
+            .collect();
+        if !needed_discussions.is_empty() {
+            tracing::info!(
+                r#"You're using some "preview" features of the Apollo Router (those which have their configuration prefixed by "preview").  These features are not officially supported with any SLA and may still contain bugs or undergo iteration. You're encouraged to try preview features in test environments to familiarize yourself with upcoming functionality before it reaches general availability. For more information about launch stages, please see the documentation here: https://www.apollographql.com/docs/resources/product-launch-stages/"#,
```

Comment:
Should this message include which preview features are used, like the corresponding message does for experimental features?

Reply:
I'm glad you caught that, too.  I was also wondering that, but I cherry picked this off another branch.  I meant to either:

1. Check it out and try out how it behaved in practice.
2. Write a test that snapshotted it.

At this moment, I'm not sure when I'll be able to get back to this PR though, but this point of uncertainty still stands right now.